src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -120,39 +122,28 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig, protected
     }
   }
 
-  private def getFirewallRule(googleProject: String, firewallRuleName: String): Future[Unit] = {
-    Future {
-      val request = compute.firewalls().get(googleProject, firewallRuleName)
-      executeGoogleRequest(request) // returns a Firewall
-      ()
-    }.recoverWith {
-      case e: GoogleJsonResponseException =>
-        if (e.getStatusCode == 404)
-          throw FirewallRuleNotFoundException(googleProject, firewallRuleName)
-        throw CallToGoogleApiFailedException(googleProject, firewallRuleName, e.getStatusCode, e.getDetails.getMessage)
-    }
+  private def getFirewallRule(googleProject: GoogleProject, firewallRuleName: String): Future[Unit] = {
```

Comment:
Changes to existing GoogleDataprocDAO methods are mainly to use `executeGoogleRequestAsync`

Reply:
Nitpick: other `get`s return real data, so I'd like this one to be consistent and move the `.void` to `updateFirewallRule`

Alternative: rename this to `checkFirewallRule` or something

Reply:
+1